2996_reflection/reflection.md

```diff
@@ -4328,13 +4334,17 @@ consteval void g(std::meta::info r, X<false> xv) {
 
   * [#.#]{.pnum} If the `$id-expression$` denotes an overload set `$S$`, overload resolution for the expression `&$S$` with no target shall select a unique function ([over.over]{.sref}); `$R$` represents that function.
 
+  * [#.#]{.pnum} Otherwise, if the `$id-expression$` denotes a variable declared by an `$init-capture$` ([expr.prim.lambda.capture]), `$R$` is ill-formed.
+
+  * [#.#]{.pnum} Otherwise, if the `$id-expression$` denotes a local entity `$E$` ([basic.pre]) for which there is an intervening enclosing `$compound-statement$` of a `$lambda-expression$` between the `$id-expression$` and the point at which `$E$` was introduced, and if `$E$` would be odr-used by a potentially-evaluated expression that names `$E$` within the innermost such `$compound-statement$`, `$R$` is ill-formed.
```

Comment:
Maybe combine the three lambda bullets?

> Otherwise, if the id-expression denotes a local entity E and
>
> * E is a variable declared by an init-capture, 
> * E is an entity captured in some intervening lambda-expression, or
> * [whatever this third case is];
>
> then R is ill-formed. 

Also I think this case here rejects this:
```cpp
void g() {
  constexpr int x = 1;
  auto f = []{
    constexpr auto r = ^^x;
    return [:r:];
  }; 
}
```
Cause like doing `int[x]` only doesn't odr-use `x` because we specifically are applying the lvalue-to-rvalue conversion to `x` and `x` is integral. 

Maybe we don't want to rely on odr-use here and instead say like... it's an error unless `E` is variable usable in constant expressions?

Reply:
Reverted this for this PR, while we hash it out.